### PR TITLE
feat(train): add auto-scaling option for Multi-GPU Training

### PR DESCRIPTION
**Pull request by nijkah**
## What this does

This PR adds an opt-in auto-scaling feature for distributed training with Accelerate. When enabled via --auto_scale=true and running with multiple processes (GPUs), LeRobot:

- Multiplies the learning rate by world size (linear LR scaling).
- Divides the number of training steps by world size (ceil-div).

This keeps total sample count roughly comparable and preserves training dynamics while improving throughput.

## How it was tested

Examples:

```bash
accelerate launch \
  --multi_gpu \
  --num_processes=2 \
  $(which lerobot-train) \
  --dataset.repo_id=${HF_USER}/my_dataset \
  --policy.type=act \
  --policy.repo_id=${HF_USER}/my_trained_policy \
  --output_dir=outputs/train/act_multi_gpu \
  --job_name=act_multi_gpu \
  --wandb.enable=true \
  --auto_scale=true
```

Tested
- [x] A6000 x 2 GPU training

### CODE REVIEW

**Comment by pkooij** [comment]
Hi thank you for this PR, very nice addition. I am very busy this week but I will take a look asap next week!

File: src/lerobot/scripts/lerobot_train.py [hunk 1]
```diff
@@ -158,6 +158,62 @@ def train(cfg: TrainPipelineConfig, accelerator: Accelerator | None = None):
     # When using accelerate, only the main process should log to avoid duplicate outputs
     is_main_process = accelerator.is_main_process
 
+    # Optionally auto-scale LR and steps when using multiple processes BEFORE logging config
+    def _scale_optimizer_lr(opt_cfg, factor: int):
```
Comment (pkooij) [inline src/lerobot/scripts/lerobot_train.py:196]: Could we place these methods in: `src/lerobot/utils/train_utils.py` ?

Comment (pkooij) [inline src/lerobot/scripts/lerobot_train.py:196]: Maybe we can call this `multi_gpu_scale_hyperparams` or something that is not tied to only lr

File: docs/source/multi_gpu_training.mdx [hunk 2]
```diff
@@ -74,44 +74,38 @@ When you launch training with accelerate:
...
-# Example: 2 GPUs with effective batch size 2x larger
-# Original: batch_size=8, steps=100000
-# With 2 GPUs: batch_size=8 (16 in total), steps=50000
-accelerate launch --num_processes=2 $(which lerobot-train) \
-  --batch_size=8 \
-  --steps=50000 \
-  --dataset.repo_id=lerobot/pusht \
-  --policy=act
-```
+Notes:
```
Comment (pkooij) [inline docs/source/multi_gpu_training.mdx:104]: Nit: call this `explanation` instead of notes

**Review by pkooij — CHANGES_REQUESTED** [review]
